File: doc/codes/python/examples/api/compute/holo/naive.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import wavelength as calc_wavelength
from autd3_pattern_holo import (
    AmplitudeTarget,
    Directivity,
    EmissionConstraint,
    NaiveOption,
    Pa,
    TransducerMask,
    naive,
)

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

center = geometry.center() + np.array([0.0, 0.0, 150.0])
foci = [
    AmplitudeTarget(
        center + np.array([-30.0, 0.0, 0.0]),
        2.5e3 * Pa,
    ),
    AmplitudeTarget(
        center + np.array([30.0, 0.0, 0.0]),
        2.5e3 * Pa,
    ),
]

wavelength = calc_wavelength(340 * m / s)
constraint = EmissionConstraint.Clamp(0x00, 0xFF)
directivity = Directivity.Sphere
mask = TransducerMask.AllEnabled
parallel = True
option = (
    # ANCHOR: option
    NaiveOption(
        constraint,
        directivity,
        mask,
        parallel,
    )
    # ANCHOR_END: option
)
dst = geometry.pattern_buffer()
# ANCHOR: api
naive(geometry, foci, wavelength, option, dst)
# ANCHOR_END: api

File: doc/codes/csharp/examples/api/compute/holo/naive.cs
using System.Numerics;
using AUTD3;
using AUTD3.Holo;
using ControlPoint = AUTD3.Holo.ControlPoint;
using static AUTD3.Units;
using static AUTD3.Holo.HoloUnits;

namespace DocSamples.ApiComputeHoloNaive;

internal static class Sample
{
    internal static void Run()
    {
        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

        var center = geometry.Center + new Vector3(0.0f, 0.0f, 150.0f);
        var foci = new[]
        {
            new ControlPoint(center + new Vector3(-30.0f, 0.0f, 0.0f), 2.5e3f * Pa),
            new ControlPoint(center + new Vector3(30.0f, 0.0f, 0.0f), 2.5e3f * Pa),
        };

        var wavelength = Pattern.Wavelength(340.0f * m / s);
        var option =
            // ANCHOR: option
            new NaiveOption(
                constraint: EmissionConstraint.Clamp(Intensity.Min, Intensity.Max),
                directivity: Directivity.Sphere,
                backend: new NalgebraBackend(),
                mask: TransducerMask.AllEnabled
            )
            // ANCHOR_END: option
            ;
        var dst = geometry.PatternBuffer();
        // ANCHOR: api
        Holo.Naive(geometry, foci, wavelength, option, dst);
        // ANCHOR_END: api
    }
}
